gfx.cpp

```diff
@@ -2037,10 +2038,38 @@ class GfxInternal
         gfx_texture.initial_resource_state_ = resource_state;
         gfx_texture.transitioned_ = false;
         gfx_texture.flags_ = flags;
+        if(resource_flags != D3D12_RESOURCE_FLAG_NONE)
+        {
+            D3D12_RESOURCE_STATES transition_state = D3D12_RESOURCE_STATE_COMMON;
+            if((resource_flags & D3D12_RESOURCE_FLAG_ALLOW_RENDER_TARGET) != 0)
+                transition_state = D3D12_RESOURCE_STATE_RENDER_TARGET;
+            else if((resource_flags & D3D12_RESOURCE_FLAG_ALLOW_DEPTH_STENCIL) != 0)
+                transition_state = D3D12_RESOURCE_STATE_DEPTH_WRITE;
+            else if((resource_flags & D3D12_RESOURCE_FLAG_ALLOW_UNORDERED_ACCESS) != 0)
+                transition_state = D3D12_RESOURCE_STATE_UNORDERED_ACCESS;
+            if(transition_state != D3D12_RESOURCE_STATE_COMMON)
+            {
+                if(transitionResource(gfx_texture, transition_state, kTransitionType_Implicit))
+                    submitPipelineBarriers();   // transition our resources if needed
+                command_list_->DiscardResource(gfx_texture.resource_, nullptr);
+            }
+            for(uint32_t i = 0; i < ARRAYSIZE(gfx_texture.dsv_descriptor_slots_); ++i)
+            {
+                gfx_texture.dsv_descriptor_slots_[i].resize(resource_desc.DepthOrArraySize);
+                for(size_t j = 0; j < gfx_texture.dsv_descriptor_slots_[i].size(); ++j)
+                    gfx_texture.dsv_descriptor_slots_[i][j] = 0xFFFFFFFFu;
+            }
+            for(uint32_t i = 0; i < ARRAYSIZE(gfx_texture.rtv_descriptor_slots_); ++i)
+            {
+                gfx_texture.rtv_descriptor_slots_[i].resize(resource_desc.DepthOrArraySize);
+                for(size_t j = 0; j < gfx_texture.rtv_descriptor_slots_[i].size(); ++j)
+                    gfx_texture.rtv_descriptor_slots_[i][j] = 0xFFFFFFFFu;
+            }
+        }
```

Comment:
When creating new textures using input resource flags the texture has to be transitioned to the corresponding resource state and the descriptor slots have to be initialised to match code in `ensureTextureHasUsageFlag`